From DB to Client #closed

BenHurCreations opened the pull request:
JS script to take the items list from the DB and display it on our website

(will be updated with more data)

BenHurCreations committed:
From DB to Client

JS script to take the item list from the DB and display it on our website

simzacks commented on script.js:
```diff
@@ -0,0 +1,17 @@
+var con = mysql.createConnection({
+  host: "localhost",
+  user: "root",
+  password: "Shaigal5%",
```
storing passwords in the repo is stronglyy discouraged.
In general, it is a good idea to not store the connection details in the source code, and this should be read from a configuration file.

BenHurCreations replied:
ok sim, how do you suggest to do it?
I'm not familliar with the process

simzacks replied:
First of all, I suggest not connecting to the database directly from the client-side javascript.
As part of your ORM, you should have a place to configure the credentials.

BenHurCreations replied:
Ok, I'd like to schedule with you and talk about how to do it best for the project
Let's coordinate in slack :)

simzacks commented on script.js:
```diff
@@ -0,0 +1,17 @@
+var mysql = require('mysql');
```
where is this javascript file going to be used? 
It looks like a serverside js, but your other code was going to be in python?

BenHurCreations replied:
it is going to be used in the website that Oded built, to display the data to the user

simzacks replied:
Generally the client doesn't speak directly to the database.
You should be using an ORM, either with flask or django which talks to the database and presents the data to the user interface.

BenHurCreations committed:
DB model for our project, including relations

- All relations are identfiying relations
- I'm not yet sure if updating the EER diagram updates automatically the tables in the model, I'll be glad for answers :)

BenHurCreations committed:
Delete script.js

BenHurCreations commented:
Irrelevant for now so I close this request
